pyacq/core/tools.py

```diff
@@ -81,28 +114,111 @@ def _initialize(self):
         #    assert 'timeaxis' in self.conversions        
-        
-        self.poller = ThreadPollInput(input_stream = self.input)
-        self.poller.new_data.connect(self.on_new_data)
+        self.thread = ThreadStreamConverter(self.input, self.output, self.conversions)
     
-    def on_new_data(self, pos, arr):
-        if 'transfermode' in self.conversions and self.conversions['transfermode'][0]=='sharedarray':
-            arr = self.input.get_array_slice(self, pos, None)
+    def _start(self):
+        self.thread.start()
 
-        if 'timeaxis' in self.conversions:
-            arr = arr.swapaxes(*self.conversions['timeaxis'])
+    def _stop(self):
+        self.thread.stop()
+        self.thread.wait()
+    
+    def _close(self):
+        pass
+
+register_node_type(StreamConverter)
+
+
+
+class ThreadStreamSplitter(ThreadPollInput):
+    """
+    
+    """
+    def __init__(self, input_stream, outputs_stream, channelaxis, nb_channel, timeout = 200, parent = None):
+        ThreadPollInput.__init__(self, input_stream, timeout = timeout, parent = parent)
+        self.outputs_stream = weakref.WeakValueDictionary()
+        self.outputs_stream.update(outputs_stream)
+        self.channelaxis = channelaxis
+        self.nb_channel = nb_channel
+    
+    def process_data(self, pos, data):
+        if self.channelaxis == 1:
+            for i in range(self.nb_channel):
+                self.outputs_stream[str(i)].send(pos, data[:, i:i+1].copy())
+        else:
+            for i in range(self.nb_channel):
+                self.outputs_stream[str(i)].send(pos, data[i:i+1, :])
+
+
+class StreamSplitter(Node):
+    """
+    StreamSplitter take a multi signal as input and split it as N single signal output.
+    
+    Work only for  transfermode = 'plaindata'.
+    
+    usage:
+    splitter = StreamSplitter()
+    splitter.configure()
+    splitter.input.connect(someinput)
+    for output in splitter.outputs.values():
+        output.configure(someotherspec)
+    splitter.initialize()
+    splitter.start()
+    
+    """
+    _input_specs = {'in' : {}}
+    _output_specs = {}#done dynamically in _configure
+    
+    def __init__(self, **kargs):
+        Node.__init__(self, **kargs)
+    
+    def _configure(self):
+        pass
+    
+    def check_input_specs(self):
+        assert self.input.params['transfermode'] == 'plaindata', 'StreamSplitter work only for transfermode=plaindata'
+
+    def check_output_specs(self):
+        for output in self.outputs.values():
+            if self.input.params['timeaxis']==0:
+                assert output.params['shape'][1] == 1, 'StreamSplitter: wrong shape'
+            else:
+                assert output.params['shape'][0] == 1, 'StreamSplitter: wrong shape'
+
+    def after_input_connect(self, inputname):
+        if self.input.params['timeaxis']==0:
+            self.channelaxis = 1
+            self.nb_channel = self.input.params['shape'][1]
+            shape = (-1, 1)
```

Comment:
But the order of axes is independent of the actual memory layout of the data..  Could we make it that streams can send  data with any axis-order, but the result is always transposed to have time=0? That way you have the freedom to choose whichever memory layout is most efficient, but the streams make it appear as if time is always axis 0.  I'll add this to my list of "maybe-do" things and let you decide if it is worthwhile.